### PR TITLE
feat(lib/fireblocks): fixes and improvements

### PR DESCRIPTION
**Pull request by DanFlannel**
Fixes and improves the fireblocks client:
 - Refactor main interface to `Sign`
 - Use proper crypto types.
 - Verify signatures against expected address
 - Add options for config
 - Add GetPubKey endpoint
 - Add smoke tests and unit tests

task: none

### CODE REVIEW

File: lib/fireblocks/createtransaction.go [hunk 1]
```diff
@@ -43,19 +43,17 @@ func newTransactionRequest(opt TransactionRequestOptions) createTransactionReque
...
-			Type: "VAULT_ACCOUNT",
-		},
 		CustomerRefID: "",
 		ExtraParameters: &extraParameters{
 			RawMessageData: rawMessageData{
+				Algorithm: "MPC_ECDSA_SECP256K1",
 				Messages: []UnsignedRawMessage{{
-					Content: hex.EncodeToString(contentHash[:]),
+					Content:        hex.EncodeToString(contentHash[:]),
+					DerivationPath: opt.Message.DerivationPath,
```
Comment (DanFlannel) [inline lib/fireblocks/createtransaction.go:56]: fails with derivation path [44, 60, 0, 0] and both signing algorithms, passes with any valid `AssetID` such as `TTTT_ETH_TEST6`

File: lib/fireblocks/jsonhttp.go [hunk 2]
```diff
@@ -41,6 +41,11 @@ func (c jsonHTTP) Send(ctx context.Context, endpoint string, httpMethod string,
 	if err != nil {
 		return errors.Wrap(err, "marshaling JSON")
 	}
+	// on get requests even will a nil request, we are passing in a non nil request body as the body marshaled to equal `null`
```
Comment (corverroos) [inline lib/fireblocks/jsonhttp.go:44]: suggest:
```
// Only marshal request if not-nil
var reqBytes []bytes
if request != nil {
	reqBytes, err = json.Marshal(request)
	if err != nil {
		return errors.Wrap(err, "marshaling JSON")
	}
}
```

File: lib/fireblocks/types.go [hunk 3]
```diff
@@ -60,8 +60,8 @@ type rawMessageData struct {
 }
 
 type UnsignedRawMessage struct {
-	Content   string `json:"content"`
-	Algorithm string `json:"algorithm,omitempty"`
+	Content        string `json:"content"`
+	DerivationPath []int  `json:"derivationPath,omitempty"`
 }
```
Comment (Lazar955) [inline lib/fireblocks/types.go:65]: Add comments

File: lib/fireblocks/createtransaction.go [hunk 4]
```diff
@@ -15,7 +15,7 @@ type TransactionRequestOptions struct {
 // CreateTransaction creates a new transaction on the FireBlocks API.
 func (c Client) CreateTransaction(ctx context.Context, opt TransactionRequestOptions) (*TransactionResponse, error) {
```
Comment (Lazar955) [inline lib/fireblocks/createtransaction.go:16]: Follow our go style guide, we don't return pointers, change to: `(TransactionResponse, error)`

File: lib/fireblocks/createandwait.go [hunk 5]
```diff
@@ -15,6 +15,15 @@ func (c Client) CreateAndWait(ctx context.Context, opts TransactionRequestOption
 		return TransactionResponse{}, err
 	}
 
+	// see if the transaction returned is already completed and exit early if it is
+	isCompleted, err := isComplete(*resp)
+	if err != nil {
+		return *resp, err
```
Comment (Lazar955) [inline lib/fireblocks/createandwait.go:21]: so we avoid pointer deref

**Review by pavelnikolov — APPROVED** [review]

**Review by corverroos — APPROVED** [review]

**Pull request merged**